main.nf

```diff
@@ -142,24 +178,85 @@ process Register_Anat {
     """ 
 }
 
+if (affine_counter.value != warp_counter.value || \
+    affine_counter.value != subj_counter.value) {
+    transfo_from_registration.set{deformation_for_warping}
+}
+else {
+    affine_for_reg
+        .join(warp_for_reg, by: 0)
+        .set{deformation_for_warping}
+}
+
+if (params.use_atlas_roi_seeding) {
+    atlas_seeds
+        .map{file -> def basename = file.name.lastIndexOf('.nii.gz') > -1 ? file.name[0..-8] : file.baseName
```

Comment:
I don't understand why would you need all of this since in atlas_seeds there are only nii.gz you can just use file.name[0..-8] without to check "nii.gz".

Reply:
The check for the extension is to get the basename, the normal basename/extension code return stuff like AF_L.nii for basename and .gz for extension.

So this is simpl